cypress/cypress.json

```diff
@@ -0,0 +1,23 @@
+{
+    "integrationFolder": "test/cypress/integration",
+    "fixturesFolder": "test/cypress/fixtures",
+    "screenshotsFolder": "test/cypress/screenshots",
+    "videosFolder": "test/cypress/videos",
+    "supportFile": "test/cypress/support/index.js",
+    "pluginsFile": "node_modules/cloudify-ui-common/cypress/plugins/index.js",
```

Comment:
I'm thinking whether it shouldn't be the other way around:
1. Each repository has its own `cypress.json` that specifies things like paths, etc.
2. In the `plugins` section of each final repository (stage, composer), there is an import `cloudify-ui-common/cypress/plugins/index.js`. That `index.js` would dynamically change Cypress's configuration and apply common settings.

This is because I have not seen any way to `extend` a `cypress.json`, as we do with ESLint or TypeScript.

If we use common plugins to extend the configuration, each repository can easily override the default configuration, if need be (e.g. do not capture videos or use a different number of retries).

https://docs.cypress.io/api/plugins/configuration-api.html#Usage

So, I would imagine that we have a single plugin in this repository that only applies the shared settings. It could also automatically add the `getAdminToken` command, or we could explicitly include it by creating a new file in `stage` and `composer` and importing `cloudify-ui-common/cypress/support/getAdminToken`. 

Then, other changes in this PR would include changes to eslint.

What do you think?

Reply:
You can still set/override any configuration options on per-repository basis with `-c` parameter passed to the runner. My plan was to use it, so for example in composer the `e2e` npm script would be extended to `cypress run -C node_modules/cloudify-ui-common/cypress/cypress.json -c baseUrl=http://localhost:9000`. In this case we don't need a per-repository plugins file.

Importing `cloudify-ui-common/cypress/support`, as per your proposal, is also part of the plan.

Reply:
I see :slightly_smiling_face: 

Adding `-c baseUrl=http://localhost:9000` to the command itself makes it harder to maintain, IMO, than a config file. If it was specified for a config file, then no matter how you run the command (whether it's `npm run cypress` or `npx cypress`), it would be picked up. If I run `npx cypress`, then I guess it will fail, since `cypress.json` will not be present in the repo

Reply:
> having a plugin that imports some other plugin is also less magical than using a config from another repo directly

OK, I can agree on this. Will make the change.

However, when it comes to cypress.json the goal was to reuse this config file as it is identical for stage/composer, except the `baseUrl` property. This property could be set programmatically making the `-c` command line parameter unnecessary, that's good, however we still need `-C` to specify non-standard cypress.json location. Without it the runner will use default configuration and will fail as it won't have correct paths set  - so unfortunately we can't set all configuration options at runtime as part of plugin initialization.

Reply:
Hmm... I can't figure out setting which shared configuration value is unachievable with plugins and has to be done with a config. Do you have any specifics in mind? :thinking:

Reply:
For sure the `pluginsFile` needs to first point to correct plugin initialization file, without it the runner will look for it in a default location, where it won't be found, so runtime configuration won't run. If so I suspect other paths may need to be set statically as well, I haven't verify it though, I'll try to check it later today.

Reply:
My intention was to have a plugins file in each repository (stage and composer), which would say:

```ts
import setCommonConfig from 'cloudify-ui-common/cypress';

module.exports = () => {
  setCommonConfig();
};
```

or something similar

Reply:
Yes, I understand, but you still need to set path to this file in the static configuration, otherwise it won't be picked up as we are using non-standard directory structure.